go-controller/pkg/crd/userdefinednetwork/v1/types.go

```diff
@@ -74,6 +74,9 @@ const (
 
 // +kubebuilder:validation:XValidation:rule="has(self.subnets) && size(self.subnets) > 0", message="Subnets is required for Layer3 topology"
 // +kubebuilder:validation:XValidation:rule="!has(self.joinSubnets) || has(self.role) && self.role == 'Primary'", message="JoinSubnets is only supported for Primary network"
+// + ---
+// + TODO: Add the following validations when available (kube v1.31).
+// + kubebuilder:validation:XValidation:rule="!has(self.subnets) || !self.subnets.exists_one(i, cidr(i.cidr).ip().family() == 6) || self.subnets.exists_one(i, cidr(i.cidr).ip().family() == 6) && self.mtu >= 1280", message="MTU should be greater than or equal to 1280 when IPv6 subent is used"
```

Comment:
I think this can be simplified to
```
!has(self.subnets) || !self.subnets.exists_one(i, cidr(i.cidr).ip().family() == 6) ||  self.mtu >= 1280
```
though I didn't try it :)

Reply:
Thanks,  I will try that and update

Reply:
It works, done.